src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -214,7 +214,7 @@ public static IEnumerable<UserDefinedFunction> CreateFunctions(IEnumerable<UDF>
                 Contracts.Assert(udf.IsParseValid);
 
                 var udfName = udf.Ident.Name;
-                if (texlFunctionSet.AnyWithName(udfName))
+                if (texlFunctionSet.AnyWithName(udfName) || nameResolver.Functions.WithName(udfName).Any(func => func is UserDefinedFunction))
                 {
                     errors.Add(new TexlError(udf.Ident, DocumentErrorSeverity.Severe, TexlStrings.ErrUDF_FunctionAlreadyDefined, udfName));
```

Comment:
>  errors.Add(new TexlError(udf.Ident, DocumentErrorSeverity.Severe, TexlStrings.ErrUDF_FunctionAlreadyDefined, udfName)); [](http://example.com/codeflow?start=18&length=121)

for collision detection, could we push this to the caller?
And not create a local set. 
   var texlFunctionSet = new TexlFunctionSet();